ui/src/services/url.ts

```diff
@@ -0,0 +1,38 @@
+const githubRegex: RegExp = /^https:\/\/github\.com\/([^/]+)\/([^/]+)\/blob\/([^/]+)\/(.+)$/;
+
+
+async function fetchUrlContent(url: string): Promise<string> {
```

Comment:
I attempted to upload a file hosted in GitLab and faced some issues.
URL: https://gitlab.com/xbaeza/primeraapi/-/raw/master/petstore.yaml
![Screenshot 2022-09-05 at 11 33 08](https://user-images.githubusercontent.com/5792097/188429306-904554a0-d73b-461e-a03a-974efa056c3d.png)

Is it expected?

Reply:
If content is not fetchable by the UI (e.g. due to CORS not being enabled on the target server) then yes it will fail.  :(  CORS is obviously not an issue when fetching from the server... so may need to reconsider this.  WDYT?

Reply:
I think that fetching from the UI is a great idea as it excludes all the security concerns, maybe we can add the fetch from the server only when this mechanism fails.
At the same time maintaining two approaches can cost in terms of maintenance ...

If I really need to provide a preference, probably, I would opt for fetching on the server side, but not fussy at all.